src/core/index.ts

```diff
@@ -257,13 +258,19 @@ export class ZarrArray {
 
   public get(selection?: undefined | Slice | ":" | "..." | null | (Slice | null | ":" | "...")[], opts?: GetOptions): Promise<NestedArray<TypedArray>>;
   public get(selection?: ArraySelection, opts?: GetOptions): Promise<NestedArray<TypedArray> | number>;
-  public get(selection: ArraySelection = null, opts: GetOptions = {}): Promise<NestedArray<TypedArray> | number> {
-    return this.getBasicSelection(selection, opts);
+  public get(selection: ArraySelection = null, opts: GetOptions = {}): Promise<NestedArray<TypedArray> | RawArray | number> {
```

Comment:
I don't think this can ever return a RawArray as we pass `false`, maybe we can change the typing of `getBasicSelection` to return a different type given the input. I'll have a stab at it in a comment below

If it doesn't end up really possible to type it correctly, we can consider casting it here

Reply:
fixed! thanks